File: spring-backend/src/main/java/com/ronpotter99/simpleschoolwebapp/service/interfaces/ClassService.java
package com.ronpotter99.simpleschoolwebapp.service.interfaces;

import com.ronpotter99.simpleschoolwebapp.entity.Class;

import java.util.List;

public interface ClassService {

    List<Class> getAllClasses();

    Class getClass(Long classId);

    List<Class> getClassesPersonTeaches(Long professorId);

    List<Class> getClassesPersonTakes(Long studentId);
}

File: angular-frontend/src/app/services/class.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class ClassService {

  private REST_API_BASE = '/api';

  constructor(private httpClient: HttpClient) { }

  public getAllClasses() {
    return this.httpClient.get(this.REST_API_BASE + '/class/all')
  }

  public getClass(classId: number) {
    return this.httpClient.get(this.REST_API_BASE + '/class/' + classId)
  }

  public getClassesPersonTeaches(professorId: number) {
    return this.httpClient.get(this.REST_API_BASE + '/class/taughtBy/' + professorId)
  }

  public getClassesPersonTakes(studentId: number) {
    return this.httpClient.get(this.REST_API_BASE + '/class/enrolledIn/' + studentId)
  }
}
